### ISSUE TITLE
local lazy val containing case class fails to compile, issue with TreeTypeMap

### ISSUE THREAD

**Issue by smarter** [lampepfl/dotty#1810]
I was asked to provide an example of `TreeTypeMap` being broken in some other phase than `ElimByName`, so here you go :).
```scala
class Test {
  def test: Unit = {
    lazy val x: Int = {
      case class Foo(x: Int)
      1
    }
  }
}
```
Fails with the mysterious:
```scala
constructor throws in class throws does not take type parameters
```
Stack trace after using `-Xprompt`: https://gist.github.com/d121067571cdcbf9bac26cf8028f6002

**Comment by DarkDimius** [comment]
I'm Digging into this.

**Comment by DarkDimius** [comment]
Fixed in #1811.
